plugins/woocommerce/src/Blocks/Utils/ProductCollectionUtils.php

```diff
@@ -92,4 +99,179 @@ private static function remove_empty_array_recursive( $array ) {
 		}
 		return $array;
 	}
+
+	/**
+	 * Parse editor's location context from WP Post.
+	 *
+	 * Possible contexts:
+	 * - post
+	 * - page
+	 * - single-product
+	 * - product-archive
+	 * - cart
+	 * - checkout
+	 * - product-catalog
+	 * - order-confirmation
+	 *
+	 * @param WP_Post $post The Post instance.
+	 *
+	 * @return string Returns the context.
+	 */
+	public static function parse_editor_location_context( $post ) {
+		$context = 'site';
+
+		if ( ! $post instanceof \WP_Post ) {
+			return $context;
+		}
+
+		$post_type = $post->post_type;
+		if ( ! in_array( $post_type, array( 'post', 'page', 'wp_template', 'wp_template_part' ), true ) ) {
+			return $context;
+		}
+
+		if ( in_array( $post_type, array( 'wp_template', 'wp_template_part' ), true ) ) {
+
+			$name = $post->post_name;
+			if ( false !== strpos( $name, SingleProductTemplate::SLUG ) ) {
+				$context = 'single-product';
+			} elseif ( ProductAttributeTemplate::SLUG === $name ) {
+				$context = 'product-archive';
+			} elseif ( false !== strpos( $name, 'taxonomy-' ) ) { // Including the '-' in the check to avoid false positives.
+				$taxonomy           = str_replace( 'taxonomy-', '', $name );
+				$product_taxonomies = get_object_taxonomies( 'product', 'names' );
+				if ( in_array( $taxonomy, $product_taxonomies, true ) ) {
+					$context = 'product-archive';
+				}
+			} elseif ( in_array( $name, array( CartTemplate::SLUG, MiniCartTemplate::SLUG ), true ) ) {
+				$context = 'cart';
+			} elseif ( CheckoutTemplate::SLUG === $name ) {
+				$context = 'checkout';
+			} elseif ( ProductCatalogTemplate::SLUG === $name ) {
+				$context = 'product-catalog';
+			} elseif ( OrderConfirmationTemplate::SLUG === $name ) {
+				$context = 'order-confirmation';
+			}
+		}
+
+		if ( 'page' === $post_type ) {
+			$context = 'page';
+		}
+		if ( 'post' === $post_type ) {
+			$context = 'post';
+		}
+
+		return $context;
+	}
+
+	/**
+	 * Track usage of the Product Collection block within the given blocks.
+	 *
+	 * @param array $blocks     The parsed blocks to check.
+	 * @param bool  $in_single  Whether we are in a single product container (used for keeping state in the recurring process).
+	 *
+	 * @return array Parsed instances of the Product Collection block.
+	 */
+	public static function parse_blocks_track_data( $blocks, $in_single = false ) {
+
+		$instances = array();
+
+		foreach ( $blocks as $block ) {
+			if ( 'woocommerce/product-collection' === $block['blockName'] ) {
+
+				$instances[] = array(
+					'collection'        => $block['attrs']['collection'] ?? 'product-catalog',
+					'in-single-product' => $in_single ? '1' : '0',
```

Comment:
Question: Is there any reason behind using `'1'` and `'0'` instead of `true` and `false`?

Reply:
Good question! 

I wouldn't say I have a tangible reason other than that specific strings are generally handled better with type checking or by using something like `wc_string_to_bool; booleans might be converted into "something" during requests

Based on this review, I'm now considering making these `yes` and `no`. Thoughts?

Reply:
I think using 'yes' and 'no' is a good idea. It will make more sense and will be more readable. So, I am in favor of using 'yes' and 'no' instead of '1' and '0'. :100: